### PR TITLE
Bring SmartOS defaults inline with current pkgsrc

### PR DESCRIPTION
**Pull request by coyhile**
### Description
Fix the following issues found on SmartOS: 

- The existing template in `templates/smartos/50-default.conf.erb` is incompatible with current rsyslog versions.
- Defaults on SmartOS enable the `imudp` and `imtcp` without a listener. (These modules are listed, but commented in the default config from pkgsrc.)

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: spec/default_spec.rb [hunk 1]
```diff
@@ -209,7 +209,7 @@
       let(:template) { chef_run.template('/opt/local/etc/rsyslog.d/50-default.conf') }
 
       it 'creates the template' do
-        expect(chef_run).to render_file(template.path).with_content('Default rules for rsyslog.')
+        expect(chef_run).to render_file(template.path).with_content('Generated by Chef')
```
Comment (tas50) [inline spec/default_spec.rb:212]:
```suggestion
        expect(chef_run).to render_file(template.path).with_content('Generated by Chef Infra Client')
```

Comment (coyhile) [reply spec/default_spec.rb:212]: This one should be dealt with in the latest commit - and I fixed the long-failing tests.

**Comment by coyhile** [comment]
Should I update the version in metadata.rb, or will whomever releases the new version do that?

**Comment by Xorima** [comment]
Hey @coyhile 

Can you rebase this?
Also you do not need to worry about the version number, we can handle that automatically

Thanks!

File: metadata.rb [hunk 2]
```diff
@@ -17,6 +17,11 @@
...
 supports 'redhat'
 supports 'scientific'
+supports 'smartos'
 supports 'suse'
 supports 'ubuntu'
 supports 'zlinux'
+
+source_url 'https://github.com/chef-cookbooks/rsyslog'
+issues_url 'https://github.com/chef-cookbooks/rsyslog/issues'
+chef_version '>= 13'
```
Comment (ramereth) [inline metadata.rb:27]: Please remove this. Looks like it was brought in when you rebased it.

**Review by ramereth — CHANGES_REQUESTED** [review]

**Review by ramereth — COMMENTED** [review]
Can you please add a Changelog entry for this? Looks like our automated check didn't quite catch that correctly (cc/ @Xorima)

**Review by ramereth — DISMISSED** [review]

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [7.6.0](https://github.com/sous-chefs/rsyslog/releases/tag/7.6.0)

**Pull request merged**